diff -ur ocaml-3.05/ocamldoc/odoc_messages.ml ocaml-3.05-/ocamldoc/odoc_messages.ml
--- ocaml-3.05/ocamldoc/odoc_messages.ml	Tue Jul 23 16:11:51 2002
+++ ocaml-3.05-/ocamldoc/odoc_messages.ml	Thu Aug  8 14:09:56 2002
@@ -63,7 +63,7 @@
 
 let man_mini = " Generate man pages only for modules, module types,\n"^
   "        classes and class types "^man_only
-let default_man_suffix = "o"
+let default_man_suffix = "3o"
 let man_suffix = "<suffix>  use <suffix> for man page files "^
   "(default is "^default_man_suffix^") "^man_only^"\n"
 
