diff -ur ocaml-3.05/ocamldoc/Makefile ocaml-3.05-/ocamldoc/Makefile
--- ocaml-3.05/ocamldoc/Makefile	Tue Jul 23 15:21:14 2002
+++ ocaml-3.05-/ocamldoc/Makefile	Thu Aug  8 14:11:16 2002
@@ -40,7 +40,7 @@
 OCAMLDOC_LIBA=odoc_info.a
 INSTALL_LIBDIR=$(OCAMLLIB)/ocamldoc
 INSTALL_BINDIR=$(OCAMLBIN)
-INSTALL_MANODIR=$(MANDIR)/mano
+INSTALL_MANODIR=$(MANDIR)/man3
 
 INSTALL_MLIS=odoc_info.mli odoc_args.mli
 INSTALL_CMIS=$(INSTALL_MLIS:.mli=.cmi)
@@ -196,7 +196,7 @@
 $(OCAMLDOC_LIBCMXA): $(LIBCMXFILES)
 	$(OCAMLOPT) -a -o $@ $(LINKFLAGS) $(OCAMLCMXFILES) $(LIBCMXFILES)
 
-manpages: stdlib_man/Pervasives.o
+manpages: stdlib_man/Pervasives.3o
 
 odoc_crc.ml: $(CMIFILES)
 	$(EXTRAC_CRC) $(INCLUDES) \
@@ -315,7 +315,7 @@
 	$(MKDIR) $@ 
 	$(OCAMLDOC_RUN) -texi -d $@ $(INCLUDES) odoc*.ml odoc*.mli
 
-stdlib_man/Pervasives.o: $(STDLIB_MLIS)
+stdlib_man/Pervasives.3o: $(STDLIB_MLIS)
 	$(MKDIR) stdlib_man
 	$(OCAMLDOC_RUN) -man -d stdlib_man $(INCLUDES) \
 	-t "OCaml library" -man-mini \
diff -ur ocaml-3.05/ocamldoc/odoc_messages.ml ocaml-3.05-/ocamldoc/odoc_messages.ml
--- ocaml-3.05/ocamldoc/odoc_messages.ml	Tue Jul 23 16:11:51 2002
+++ ocaml-3.05-/ocamldoc/odoc_messages.ml	Thu Aug  8 14:09:56 2002
@@ -63,7 +63,7 @@
 
 let man_mini = " Generate man pages only for modules, module types,\n"^
   "        classes and class types "^man_only
-let default_man_suffix = "o"
+let default_man_suffix = "3o"
 let man_suffix = "<suffix>  use <suffix> for man page files "^
   "(default is "^default_man_suffix^") "^man_only^"\n"
 
