--- ocaml-3.08.3/configure.orig	2005-04-29 22:58:18.000000000 +0200
+++ ocaml-3.08.3/configure	2005-04-29 23:06:30.000000000 +0200
@@ -1322,6 +1322,7 @@
     8.2) tclmaj=8 tclmin=2 tkmaj=8 tkmin=2 ;;
     8.3) tclmaj=8 tclmin=3 tkmaj=8 tkmin=3 ;;
     8.4) tclmaj=8 tclmin=4 tkmaj=8 tkmin=4 ;;
+    8.5) tclmaj=8 tclmin=5 tkmaj=8 tkmin=5 ;;
     *) echo "This version is not known."; has_tk=false ;;
     esac
   else
