--- ocaml-3.06/otherlibs/labltk/support/cltkImg.c.orig	2002-07-23 16:11:59.000000000 +0200
+++ ocaml-3.06/otherlibs/labltk/support/cltkImg.c	2003-06-19 03:06:34.000000000 +0200
@@ -99,7 +99,7 @@
   pib.offset[0] = 0;
   pib.offset[1] = 1;
   pib.offset[2] = 2;
-  Tk_PhotoPutBlock(ph,&pib,Int_val(x),Int_val(y),Int_val(w),Int_val(h)); 
+  Tk_PhotoPutBlock_NoComposite(ph,&pib,Int_val(x),Int_val(y),Int_val(w),Int_val(h)); 
 }
 
 CAMLprim void camltk_setimgdata_bytecode(argv,argn)
--- ocaml-3.06/otherlibs/labltk/tkanim/tkAnimGIF.c.orig	2002-07-23 16:12:00.000000000 +0200
+++ ocaml-3.06/otherlibs/labltk/tkanim/tkAnimGIF.c	2003-06-19 03:17:18.000000000 +0200
@@ -335,7 +335,7 @@
                 goto error;
             }
         }
-        Tk_PhotoPutBlock(photoHandle, &block, 0, 0, 
+        Tk_PhotoPutBlock_NoComposite(photoHandle, &block, 0, 0, 
                          imageWidth, imageHeight);
 #ifdef TKANIM_DEBUG
     fprintf(stderr, " Retrieving result\n");
