--- ocaml-3.12.1/configure~	2014-06-03 11:26:08.000000000 +0200
+++ ocaml-3.12.1/configure	2014-06-03 11:27:37.713580505 +0200
@@ -1429,6 +1429,8 @@
   for tk_incs in \
     "-I/usr/local/include" \
     "-I/usr/include" \
+    "-I/usr/local/include/tcl8.6 -I/usr/local/include/tk8.6" \
+    "-I/usr/include/tcl8.6 -I/usr/include/tk8.6" \
     "-I/usr/local/include/tcl8.5 -I/usr/local/include/tk8.5" \
     "-I/usr/include/tcl8.5 -I/usr/include/tk8.5" \
     "-I/usr/local/include/tcl8.4 -I/usr/local/include/tk8.4" \
@@ -1454,6 +1456,7 @@
     8.3) tclmaj=8 tclmin=3 tkmaj=8 tkmin=3 ;;
     8.4) tclmaj=8 tclmin=4 tkmaj=8 tkmin=4 ;;
     8.5) tclmaj=8 tclmin=5 tkmaj=8 tkmin=5 ;;
+    8.6) tclmaj=8 tclmin=6 tkmaj=8 tkmin=6 ;;
     *) echo "This version is not known."; has_tk=false ;;
     esac
   else
