diff -urN ocaml-3.10.0.org/configure ocaml-3.10.0/configure
--- ocaml-3.10.0.org/configure	2007-05-03 14:49:56.000000000 +0200
+++ ocaml-3.10.0/configure	2007-07-15 20:20:19.325848428 +0200
@@ -1252,51 +1252,13 @@
 
 # See if we can compile the dbm library
 
-dbm_include="not found"
-dbm_link="not found"
 use_gdbm_ndbm=no
 
-for dir in /usr/include /usr/include/db1 /usr/include/gdbm; do
-  if test -f $dir/ndbm.h; then
-    dbm_include=$dir
-    if sh ./hasgot dbm_open; then
-      dbm_link=""
-    elif sh ./hasgot -lndbm dbm_open; then
-      dbm_link="-lndbm"
-    elif sh ./hasgot -ldb1 dbm_open; then
-      dbm_link="-ldb1"
-    elif sh ./hasgot -lgdbm dbm_open; then
-      dbm_link="-lgdbm"
-    elif sh ./hasgot -lgdbm_compat -lgdbm dbm_open; then
-      dbm_link="-lgdbm_compat -lgdbm"
-    fi
-    break
-  fi
-  if test -f $dir/gdbm-ndbm.h; then
-    dbm_include=$dir
-    use_gdbm_ndbm=yes
-    if sh ./hasgot -lgdbm_compat -lgdbm dbm_open; then
-      dbm_link="-lgdbm_compat -lgdbm"
-    fi
-    break
-  fi
-done
-if test "$dbm_include" = "not found" || test "$dbm_link" = "not found"; then
-  echo "NDBM not found, the \"dbm\" library will not be supported."
-else
-  echo "NDBM found (in $dbm_include)"
-  if test "$dbm_include" = "/usr/include"; then
-    dbm_include=""
-  else
-    dbm_include="-I$dbm_include"
-  fi
-  if test "$use_gdbm_ndbm" = "yes"; then
-    echo "#define DBM_USES_GDBM_NDBM" >> s.h
-  fi
-  otherlibraries="$otherlibraries dbm"
-fi
+dbm_include=""
+dbm_link="-ldb"
 echo "DBM_INCLUDES=$dbm_include" >> Makefile
 echo "DBM_LINK=$dbm_link" >> Makefile
+otherlibraries="$otherlibraries dbm"
 
 # Look for tcl/tk
 
diff -ur ocaml-3.07beta2/otherlibs/dbm/cldbm.c ocaml-3.07beta2-/otherlibs/dbm/cldbm.c
--- ocaml-3.07beta2/otherlibs/dbm/cldbm.c	2003-07-08 15:50:31.000000000 +0200
+++ ocaml-3.07beta2-/otherlibs/dbm/cldbm.c	2003-08-26 11:11:19.000000000 +0200
@@ -24,7 +24,8 @@
 #ifdef DBM_USES_GDBM_NDBM
 #include <gdbm-ndbm.h>
 #else
-#include <ndbm.h>
+#define DB_DBM_HSEARCH 1
+#include <db.h>
 #endif
 
 /* Quite close to sys_open_flags, but we need RDWR */
