--- ocaml-3.08.0/ocamldoc/odoc_messages.ml	2004-07-09 16:32:42.000000000 +0200
+++ ocaml-3.08.0_/ocamldoc/odoc_messages.ml	2004-07-21 21:13:04.724002928 +0200
@@ -83,7 +83,7 @@
 
 let man_mini = "\tGenerate man pages only for modules, module types, classes\n"^
   "\t\tand class types "^man_only
-let default_man_suffix = "o"
+let default_man_suffix = "3o"
 let man_suffix = "<suffix>\n\t\tUse <suffix> for man page files "^
   "(default is "^default_man_suffix^") "^man_only^"\n"
 
